tck/src/main/java/org/eclipse/microprofile/config/tck/EmptyValue/EmptyValuesTestProgrammaticLookup.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.tck.EmptyValue;
+
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+import org.eclipse.microprofile.config.Config;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+
+public class EmptyValuesTestProgrammaticLookup extends Arquillian {
+
+    @Deployment
+    public static Archive deployment() {
+        JavaArchive jar = ShrinkWrap.create(JavaArchive.class, "emptyValues.jar")
+                .addClasses(EmptyValuesTest.class, EmptyValuesBean.class)
+                .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml")
+                .addAsManifestResource(new StringAsset(
+                                            "empty.string=" + "\n" +
+                                            "comma.string=," + "\n" +
+                                            "backslach.comma.string=\\," + "\n" +
+                                            "double.comma.string=,," + "\n" +
+                                            "foo.bar.string=foo,bar"+ "\n" +
+                                            "foo.comma.string=foo,"+ "\n" +
+                                            "comma.bar.string=,bar" + "\n" +
+                                            "space.string= "
+                                            ),
+                                       "microprofile-config.properties");
+
+        return ShrinkWrap.create(WebArchive.class)
+                .addAsLibrary(jar)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @Inject
+    private Config config;
+
+    @Test(expectedExceptions = NoSuchElementException.class)
+    public void testMissingStringGetValueArray() {
+        String[] values = config.getValue("missing.string", String[].class);
```

Comment:
In such cases I like to create a helper, like `void assertConfigurationNotPresent(String property);` which would use all the different ways to check that so that you get better readability and more coverage as you will cover the matrix with different ways properties are defined and different ways they are resolved.

Reply:
good idea. done